internal/streamingnode/server/flusher/flusherimpl/flusher_impl.go

```diff
@@ -126,11 +125,19 @@ func (f *flusherImpl) notify() {
 }
 
 func (f *flusherImpl) Start() {
-	f.stopWg.Add(1)
 	f.wbMgr.Start()
-	go f.cpUpdater.Start()
 	go func() {
-		defer f.stopWg.Done()
+		defer f.notifier.Finish(struct{}{})
+		dc, err := resource.Resource().DataCoordClient().GetWithContext(f.notifier.Context())
+		if err != nil {
+			return
+		}
+		broker := broker.NewCoordBroker(dc, paramtable.GetNodeID())
+		cpUpdater := util.NewChannelCheckpointUpdater(broker)
+		go cpUpdater.Start()
+		// When the flusher exits, the cpUpdater should be closed.
```

Comment:
Could this result in being unable to update the checkpoint when the flowgraphs are closing?

Reply:
Got it